### DISCUSSION TITLE
In which order are errors evaluated?

### DISCUSSION THREAD

**Discussion by mattgabor** [react-hook-form/react-hook-form#2111] [Q&A] [answered]
For handling multiple errors, I saw that in #1871 we can set `validateCriteriaMode: "all"` to retrieve all errors for a field. However, I was wondering what order the errors are evaluated in when `validateCriteriaMode` is set to firstError? Currently I'm using `required` and `pattern`, and I'd expect `required` to trigger before `pattern` but the opposite is happening.

**Comment by bluebill1049** [comment, marked as answer]
we don't guarantee the order of build-in validation. (as what user-defined)

**Reply by bluebill1049** [reply]
you can up vote this feature request: https://github.com/react-hook-form/react-hook-form/issues/1387
